src/android/FileTransfer.java

```diff
@@ -528,14 +528,18 @@ public void run() {
                         }
                         safeClose(inStream);
                     }
-                    
+
                     Log.d(LOG_TAG, "got response from server");
                     Log.d(LOG_TAG, responseString.substring(0, Math.min(256, responseString.length())));
-                    
+
                     // send request and retrieve response
                     result.setResponseCode(responseCode);
                     result.setResponse(responseString);
 
+                    if(result.getBytesSent() != totalBytes) {
```

Comment:
Is this code related to JIRA, this PR fixes? If no, consider submitting it via separate PR.